tungsten_ci_utils/nexus_cleanup/docker_cleanup.groovy

```diff
@@ -0,0 +1,94 @@
+    log.info("Error: " + e);
+} finally {
+    if (tx != null)
+        tx.close();
+}
+
+if (components != null) {
+    def retentionDate = DateTime.now().minusDays(retentionDays).dayOfMonth().roundFloorCopy();
+    int deletedComponentCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+
+// comp.version() expected to be in a format {{ os }}-{{ realse }}-{{ branch }}-{{ build }} e.g. rhel-queens-master-386 
+
+    listOfComponents.reverseEach { comp ->
+        def tag = comp.version();
+        def tagSplited = tag.split("-");
+        log.info("tagsplit ${tag} ${tagSplited}");
+        def build_number = tagSplited[tagSplited.length - 1];
+        // two_parts_build_number is checking correct of build_number with whitelist suffixes
+        // covering build number like `5.0.3-0.310` and also different branches from whitelist 5.0 vs 5.1
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + tagSplited[tagSplited.length - 1];
+        log.info("new two parts ${two_parts_build_number}");
+        if (whitelisted_tag_suffixes.contains(build_number)) {
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
+        } else if (whitelisted_tag_suffixes.contains(two_parts_build_number)){
+            log.info("Component skipped: ${comp.name()} ${comp.version()}");
+        }else {
+            if (!tagList.contains(build_number.toInteger())) {
+                tagList.add(build_number.toInteger());
+            }
+        }
+    }
+    if(tagList.size() > retentionCount){
+        retentionList = tagList.subList(0, tagList.size() - retentionCount);
+        listOfComponents.reverseEach { comp ->
+        def tag = comp.version();
+        def tagSplited = tag.split("-");
+        def build_number = tagSplited[tagSplited.length - 1];
+        def two_parts_build_number = tagSplited[tagSplited.length - 2] + "-" + tagSplited[tagSplited.length - 1];
+        def retentionList = tagList;
+        
+        if (!whitelisted_tag_suffixes.contains(build_number)) {
+            if (!whitelisted_tag_suffixes.contains(two_parts_build_number)) {
+                if (retentionList.contains(build_number.toInteger())) {
+                    if (comp.lastUpdated() < retentionDate) {
+                        log.info("retentionDate: ${comp.lastUpdated()} isOlder ${retentionDate}");
+                        log.info("deleting ${comp.name()}, version: ${comp.version()}");
+                        // uncomment to delete components and their assets
+                        // service.deleteComponent(repo, comp);
+                        log.info("----------");
+                        deletedComponentCount++;
+                    } else {
+                        log.info("Component skipped: ${comp.name()} ${comp.version()}");
+                    }
+                }
+            }
+        }
+    }
+        log.info("\n\nDeleted Component count: ${deletedComponentCount} \n");
```

Comment:
Please make use of formatting IDE's. This line looks weird between two right parentheses. The parenthesis in the line above matches the one which opens the closure in line 65. And the `listOfComponents` is not in the same column as the parenthesis in line 88.